src/m365/spfx/commands/project/project-externalize/rules/DynamicRule.ts

```diff
@@ -17,50 +17,57 @@ export class DynamicRule extends BasicDependencyRule {
   private restrictedNamespaces = ['@types/', '@microsoft/'];
   private fileVariationSuffixes = ['.min', '.bundle', '-min', '.bundle.min'];
 
-  public visit(project: Project): Promise<VisitationResult> {
+  public async visit(project: Project): Promise<VisitationResult> {
     if (!project.packageJson ||
       !project.packageJson.dependencies) {
-      return Promise.resolve({ entries: [], suggestions: [] });
+      return { entries: [], suggestions: [] };
     }
 
     const validPackageNames: string[] = Object.getOwnPropertyNames(project.packageJson.dependencies)
       .filter(x => this.restrictedNamespaces.map(y => x.indexOf(y) === -1).reduce((y, z) => y && z))
       .filter(x => this.restrictedModules.indexOf(x) === -1);
 
-    return Promise
-      .all(validPackageNames.map((x) => this.getExternalEntryForPackage(x, project)))
-      .then((res: (ExternalizeEntry | undefined)[]) => {
-        return {
-          entries: res
-            .filter(x => x !== undefined)
-            .map(x => x as ExternalizeEntry),
-          suggestions: []
-        };
-      });
+    const results = [];
+    for (const packageName of validPackageNames) {
```

Comment:
Hé @nicodecleyre, you should not refactor a promise.all to a serial promise execution... in such a case you can await the promise.all

Reply:
Hi @nicodecleyre, I meant keep the promise.all, but await it... so remove the promise chain. 😂 if you can fix that we can merge it.